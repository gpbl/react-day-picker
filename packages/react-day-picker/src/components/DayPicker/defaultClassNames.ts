import { DayPickerClassNames } from 'types';

const classNames: DayPickerClassNames = {
  /** Root element */
  root: 'rdp',
  caption: 'rdp-caption',

  // Day Component
  day: 'rdp-day',
  dayWrapper: 'rdp-day_wrapper',

  // Month Component
  month: 'rdp-month',
  monthTable: 'rdp-table',
  monthTbody: 'rdp-body',
  months: 'rdp-months',

  // Head Components
  head: 'rdp-head',
  headRow: 'rdp-head-row',
  headWeekNumber: 'rdp-head_weeknumber',
  headWeekName: 'rdp-head-cell',

  // Navigation Component
  nav: 'rdp-nav',
  navPrev: 'rdp-nav_prev',
  navNext: 'rdp-nav_next',

  // Week Component
  week: 'rdp-row',
  weekDay: 'rdp-cell',
  weekWeeknumber: 'rdp-row_head',

  // WeekNumber component
  weekNumber: 'rdp-rownumber',

  // Modifiers
  selected: 'rdp-day_selected',
  disabled: 'rdp-day_disabled',
  today: 'rdp-day_today',
  aftermonth: 'rdp-day_outside',
  beforemonth: 'rdp-day_outside'
};

export default classNames;
