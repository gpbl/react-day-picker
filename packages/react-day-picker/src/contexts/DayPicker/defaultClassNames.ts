import { ClassNames } from 'types/Styles';

/**
 * The name of the default CSS classes.
 */
export const defaultClassNames: Required<ClassNames> = {
  root: 'rdp',
  multiple_months: 'rdp-multiple_months',
  with_weeknumber: 'rdp-with_weeknumber',
  vhidden: 'rdp-vhidden',
  button_reset: 'rdp-button_reset',
  button: 'rdp-button',

  caption: 'rdp-caption',

  caption_start: 'rdp-caption_first',
  caption_end: 'rdp-caption_end',
  caption_between: 'rdp-caption_between',
  caption_label: 'rdp-caption_label',

  caption_dropdowns: 'rdp-caption_dropdowns',

  dropdown: 'rdp-dropdown',
  dropdown_month: 'rdp-dropdown_month',
  dropdown_year: 'rdp-dropdown_year',
  dropdown_icon: 'rdp-dropdown_icon',

  months: 'rdp-months',
  month: 'rdp-month',
  table: 'rdp-table',
  tbody: 'rdp-tbody',
  tfoot: 'rdp-tfoot',

  head: 'rdp-head',
  head_row: 'rdp-head_row',
  head_cell: 'rdp-head_cell',

  nav: 'rdp-nav',
  nav_button: 'rdp-nav_button',
  nav_button_previous: 'rdp-nav_button_previous',
  nav_button_next: 'rdp-nav_button_next',

  nav_icon: 'rdp-nav_icon',

  row: 'rdp-row',
  weeknumber: 'rdp-weeknumber',
  cell: 'rdp-cell',

  day: 'rdp-day',
  day_outside: 'rdp-day_outside'
};
