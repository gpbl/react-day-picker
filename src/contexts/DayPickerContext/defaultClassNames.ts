import type { ClassNames } from '../../types/styles';

/**
 * The name of the default CSS classes.
 */
export const defaultClassNames: Required<ClassNames> = {
  rdp: 'rdp',
  multiple_months: 'multiple_months',
  with_weeknumber: 'with_weeknumber',

  month_caption: 'month_caption',

  caption_start: 'caption_start',
  caption_end: 'caption_end',
  caption_between: 'caption_between',
  caption_label: 'caption_label',

  caption_dropdowns: 'caption_dropdowns',

  dropdown: 'dropdown',
  dropdown_month: 'dropdown_month',
  dropdown_year: 'dropdown_year',
  dropdown_icon: 'dropdown_icon',

  months_wrapper: 'months_wrapper',
  month_grid_wrapper: 'month_grid_wrapper',
  month_grid: 'month_grid',
  month_rowgroup: 'month_rowgroup',

  weekdays_row: 'weekdays_row',
  weekday_columnheader: 'weekday_columnheader',

  week_row: 'week_row',

  weeknumber_rowheader: 'weeknumber_rowheader',

  nav: 'nav',
  button_previous: 'button_previous',
  button_next: 'button_next',

  button_icon: 'button_icon',

  day: 'day',
  day_today: 'day_today',
  day_outside: 'day_outside',
  day_selected: 'day_selected',
  day_disabled: 'day_disabled',
  day_hidden: 'day_hidden',
  day_range_start: 'day_range_start',
  day_range_end: 'day_range_end',
  day_range_middle: 'day_range_middle',

  caption: 'caption',
  head: 'head',
  head_row: 'head_row',
  row: 'row',
  head_cell: 'head_cell',
  cell: 'cell',
  footer: 'footer',
  contrast_more: 'contrast_more',
  contrast_less: 'contrast_less',
  color_dark: 'dark',
  color_light: 'light',
  color_auto: '',
  'contrast_no-preference': ''
};
