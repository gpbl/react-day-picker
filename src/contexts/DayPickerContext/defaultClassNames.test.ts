import { defaultClassNames } from './defaultClassNames';

test('should match the snapshot', () => {
  expect(defaultClassNames).toMatchSnapshot();
});
