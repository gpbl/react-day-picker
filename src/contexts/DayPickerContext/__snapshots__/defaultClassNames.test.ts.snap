// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
{
  "button_icon": "button_icon",
  "button_next": "button_next",
  "button_previous": "button_previous",
  "caption": "caption",
  "caption_between": "caption_between",
  "caption_dropdowns": "caption_dropdowns",
  "caption_end": "caption_end",
  "caption_label": "caption_label",
  "caption_start": "caption_start",
  "cell": "cell",
  "color_auto": "",
  "color_dark": "dark",
  "color_light": "light",
  "contrast_less": "contrast_less",
  "contrast_more": "contrast_more",
  "contrast_no-preference": "",
  "day": "day",
  "day_disabled": "day_disabled",
  "day_hidden": "day_hidden",
  "day_outside": "day_outside",
  "day_range_end": "day_range_end",
  "day_range_middle": "day_range_middle",
  "day_range_start": "day_range_start",
  "day_selected": "day_selected",
  "day_today": "day_today",
  "dropdown": "dropdown",
  "dropdown_icon": "dropdown_icon",
  "dropdown_month": "dropdown_month",
  "dropdown_year": "dropdown_year",
  "footer": "footer",
  "head": "head",
  "head_cell": "head_cell",
  "head_row": "head_row",
  "month_caption": "month_caption",
  "month_grid": "month_grid",
  "month_grid_wrapper": "month_grid_wrapper",
  "month_rowgroup": "month_rowgroup",
  "months_wrapper": "months_wrapper",
  "multiple_months": "multiple_months",
  "nav": "nav",
  "rdp": "rdp",
  "row": "row",
  "week_row": "week_row",
  "weekday_columnheader": "weekday_columnheader",
  "weekdays_row": "weekdays_row",
  "weeknumber_rowheader": "weeknumber_rowheader",
  "with_weeknumber": "with_weeknumber",
}
`;
